iphone/cli/commands/_build.js

```diff
@@ -2226,30 +2227,30 @@ iOSBuilder.prototype.validate = function validate(logger, config, cli) {
 								module.libName = xcFrameworkOfLib;
 								module.libFile = path.join(module.modulePath, module.libName);
 								module.isXCFrameworkOfLib = true;
-
-								// TODO: read Info.plist to get the full scope of targets/arches supported!
-								let archDir = 'ios-arm64_i386_x86_64-simulator';
-								if (!fs.existsSync(path.join(module.libFile, archDir))) {
-									// Try XCode 11 dir w/o arm64 support
-									archDir = 'ios-i386_x86_64-simulator';
-									this.legacyModules.add(module.id);// Record that this won't support arm64 sim!
+								const xcFrameworkInfo = plist.readFileSync(path.join(module.libFile, 'Info.plist'));
+
+								for (const libInfo of xcFrameworkInfo.AvailableLibraries) {
+									if (libInfo.SupportedPlatformVariant === undefined) {
+										// Device library is used for hash calculation. 
```

Comment:
<!--
  1 failure:  iphone/cli/comman...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_build.js;
  Line: 2234;
-->
- :no_entry_sign: iphone/cli/commands/_build.js line 2234 – Trailing spaces not allowed. (no-trailing-spaces)